crates/release_plz_core/src/command/release.rs

```diff
@@ -631,13 +631,6 @@ async fn release_package_if_needed(
 ) -> anyhow::Result<Option<PackageRelease>> {
     let git_tag = project.git_tag(&package.name, &package.version.to_string())?;
     let release_name = project.release_name(&package.name, &package.version.to_string())?;
-    if repo.tag_exists(&git_tag)? {
-        info!(
-            "{} {}: Already published - Tag {} already exists",
-            package.name, package.version, &git_tag
-        );
-        return Ok(None);
-    }
```

Comment:
if we publish to the registry first, we can keep this check and skip checking if the package is already published in the registry.
We could of course still check if the release was created.